File: model/src/main/java/io/github/chriszhong/model/organization/event/AttributeEventFactory.java
package io.github.chriszhong.model.organization.event;

import io.github.runtimemodels.chazm.entity.Attribute;

/**
 * The {@linkplain AttributeEventFactory} interface defines the API for constructing {@linkplain AttributeEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface AttributeEventFactory {

	/**
	 * Constructs an {@linkplain AttributeEvent}.
	 *
	 * @param category
	 *            the {@linkplain EventCategory}.
	 * @param attribute
	 *            the {@linkplain Attribute}.
	 *
	 * @return a {@linkplain AttributeEvent}.
	 */
	AttributeEvent build(EventCategory category, Attribute attribute);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/AttributeEventFactory.kt
package runtimemodels.chazm.model.entity

import runtimemodels.chazm.api.entity.Attribute
import runtimemodels.chazm.model.event.EventType

/**
 * The [AttributeEventFactory] interface defines the API for constructing [AttributeEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface AttributeEventFactory {
    /**
     * Constructs an [AttributeEvent].
     *
     * @param category  the [EventType].
     * @param attribute the [Attribute].
     * @return a [AttributeEvent].
     */
    fun build(category: EventType, attribute: Attribute): AttributeEvent
}
